### DISCUSSION TITLE
how to get all shapes when selection

### DISCUSSION THREAD

**Discussion by biotech7** [epam/ketcher#1602] [General]
Hi, Epam team!
when selecting all shapes(structures, texts, arrows....etc) or substructure, how to get all shapes(as a list) or substructure ? is there any api for this method?
Regards!

**Comment by AndreiMazol** [comment]
Hi @biotech7 ,

please use the following code to get selected objects as a new 'Struct' instance:

`const selectedStruct = ketcher.editor.structSelected()`

'Struct' object has all required properties (texts, simpleObjects, rxnArrows  & etc).

Regards,
Ketcher team

**Comment by biotech7** [comment]
Hi, @AndreiMazol,
    I'll have a try on it and hope it should work well.
